src/transaction/log_replication_atomic.cpp

```diff
@@ -402,37 +402,72 @@ namespace cublog
 	assert_release (false);
       }
 
-#if !defined (NDEBUG)
-    /* there will be no duplicated object in the map */
-    auto [begin, end] = m_locked_classes.equal_range (trid);
-    if (std::count_if (begin, end, [classoid] (const auto &it)
-    {
-      return it.second == *classoid;
-    }) > 0)
-    {
-      assert (false);
-    }
-#endif
+    m_locked_classes[trid].emplace (*classoid);
+  }
+
+  void
+  atomic_replicator::bookkeep_classname (cubthread::entry &thread_entry, const OID *classoid)
+  {
+    char *classname = NULL;
+
+    assert (!OID_ISNULL (classoid) && !OID_ISTEMP (classoid));
 
-    m_locked_classes.emplace (trid, *classoid);
+    (void) heap_get_class_name (&thread_entry, classoid, &classname);
+    if (classname != NULL)
+      {
+	m_classname_map.emplace (*classoid, classname);
```

Comment:
I think the `char *` can be moved into an `std::string`:
```suggestion
	m_classname_map.emplace (*classoid, std::move (std::string { classname } ));
```
.. and no need to `free_and_init` below.

Reply:
Thanks